packages/@aws-cdk/aws-msk/lib/cluster.ts

```diff
@@ -1,34 +1,776 @@
+   */
+  readonly key?: kms.IKey;
+}
+
+/**
+ * TLS authentication properties
+ */
+export interface TlsAuthProps {
+  /**
+   * List of ACM Certificate Authorities to enable TLS authentication.
+   *
+   * @default - none
+   */
+  readonly certificateAuthorityArns?: string[];
+}
+
+/**
+ * Configuration properties for client authentication.
+ */
+export class ClientAuthentication {
+  /**
+   * SASL authentication
+   */
+  public static sasl(props: SaslAuthProps): ClientAuthentication {
+    return new ClientAuthentication(props, undefined);
+  }
+
+  /**
+   * TLS authentication
+   */
+  public static tls(props: TlsAuthProps): ClientAuthentication {
+    return new ClientAuthentication(undefined, props);
+  }
+
+  /**
+   * @param saslProps - properties for SASL authentication
+   * @param tlsProps - properties for TLS authentication
+   */
+  private constructor(
+    public readonly saslProps?: SaslAuthProps,
+    public readonly tlsProps?: TlsAuthProps,
+  ) {}
+}
+
+/**
+ * Create a MSK Cluster.
+ *
+ * @resource AWS::MSK::Cluster
+ */
+export class Cluster extends ClusterBase {
+  /**
+   * Reference an existing cluster, defined outside of the CDK code, by name.
+   */
+  public static fromClusterArn(
+    scope: constructs.Construct,
+    id: string,
+    clusterArn: string,
+  ): ICluster {
+    class Import extends ClusterBase {
+      public readonly clusterArn = clusterArn;
+      public readonly clusterName = clusterArn.split('/')[1]; // ['arn:partition:kafka:region:account-id', clusterName, clusterId]
+    }
+
+    if (core.Token.isUnresolved(clusterArn)) {
+      throw new Error('Cannot use a token when importing a cluster using an ARN.');
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly clusterArn: string;
+  public readonly clusterName: string;
+  /** Key used to encrypt SASL/SCRAM users */
+  public readonly saslScramAuthenticationKey?: kms.IKey;
+  private _clusterDescription?: cr.AwsCustomResource;
+  private _clusterBootstrapBrokers?: cr.AwsCustomResource;
+
+  constructor(scope: constructs.Construct, id: string, props: ClusterProps) {
+    super(scope, id, {
+      physicalName: props.clusterName,
+    });
+
+    const subnetSelection = props.vpc.selectSubnets(props.vpcSubnets);
+
+    this._connections = new ec2.Connections({
+      securityGroups: props.securityGroups ?? [
+        new ec2.SecurityGroup(this, 'SecurityGroup', {
+          description: 'MSK security group',
+          vpc: props.vpc,
+        }),
+      ],
+    });
+
+    if (subnetSelection.subnets.length < 2) {
+      core.Annotations.of(this).addError(
+        `Cluster requires at least 2 subnets, got ${subnetSelection.subnets.length}`,
+      );
+    }
+
+    if (
+      !core.Token.isUnresolved(props.clusterName) &&
+      !/^[a-zA-Z0-9]+$/.test(props.clusterName) &&
+      props.clusterName.length > 64
+    ) {
+      core.Annotations.of(this).addError(
```

Comment:
Just throw instead of the annotation. Error annotations are normally used for validating lookup properties, i.e ones that can change outside of the synthesis process, this cannot. 

There are a couple of these around here, make sure you replace all of them.